cordapp-example/contracts-kotlin/src/main/kotlin/com/example/contract/CashContract.kt

```diff
@@ -18,14 +18,25 @@ class CashContract : Contract {
 
         when (command.value) {
             is Commands.Create -> requireThat {
-                // everyone can print/crate cash... because of covi... crisis
-                val cashState = tx.inputsOfType<CashState>().single()
-                "Cash value must be positive" using (cashState.balance > 0)
+                "There should be no input." using tx.inputsOfType<CashState>().isEmpty()
+                // in this exercise I allow anyone to create cash
+                val cashState = tx.outputsOfType<CashState>().single()
+                "Cash value must be positive" using (cashState.value > 0)
+                "Expect one signature" using (command.signers.size == 1)
+                "Creator must be a signer" using (command.signers.single() == cashState.owner.owningKey)
             }
             is Commands.Move -> requireThat {
-                // yet another dummy verification
-                val cashState = tx.inputsOfType<CashState>().first()
-                "Cash value must be positive" using (cashState.balance > 0)
+                "There should be one input." using (tx.inputsOfType<CashState>().size == 1)
+                "There should be one output." using (tx.outputsOfType<CashState>().size == 1)
```

Comment:
I think `== 1` is too restrictive. I would say `>= 1`. If I pay with 2 coins of 1, it is as good as with 1 coin of 2.

Reply:
good point, 
but in this very simple implementation there will be always one.

Reply:
This is the contract. The contract should delineate as precisely as possible what is allowed and what is not. I understand that it is not easy to fish states with the right amount in the vault, but that is a flow difficulty that can be left for later. You should not need to modify your contract when you ability with flows improves.

Reply:
> You should not need to modify your contract when you ability with flows improves.

I need to remember that!